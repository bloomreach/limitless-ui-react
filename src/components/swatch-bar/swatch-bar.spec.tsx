import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import { SwatchBar } from './';

describe('SwatchBar', () => {
  it('should render successfully', () => {
    const { container: { firstElementChild } } = render(
      <SwatchBar.Root>
        <SwatchBar.SwatchGroup>
          <SwatchBar.SwatchColor value="crimson" color="crimson" disabled />
          <SwatchBar.SwatchColor value="rebeccapurple" color="rebeccapurple" />
          <SwatchBar.SwatchColor value="gold" color="gold" />
        </SwatchBar.SwatchGroup>
      </SwatchBar.Root>
    );

    expect(firstElementChild).toMatchSnapshot();
  });
});
