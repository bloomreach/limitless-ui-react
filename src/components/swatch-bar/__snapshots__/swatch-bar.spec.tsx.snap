// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SwatchBar > should render successfully 1`] = `
<div
  aria-orientation="horizontal"
  class="lcui-sbar"
  data-orientation="horizontal"
  dir="ltr"
  role="toolbar"
  style="outline-color: none; outline-style: none; outline-width: initial;"
  tabindex="0"
>
  <div
    class="lcui-sbar-sg"
    data-orientation="horizontal"
    dir="ltr"
    role="group"
  >
    <button
      aria-checked="false"
      aria-label="crimson"
      class="lcui-swatch-item lcui-swatch-item--color"
      data-disabled=""
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="off"
      disabled=""
      role="radio"
      style="--lcui-swatch-btn-border-color--active: crimson;"
      tabindex="-1"
      type="button"
    >
      <div
        class="lcui-swatch-color"
        style="background-color: crimson;"
      />
    </button>
    <button
      aria-checked="false"
      aria-label="rebeccapurple"
      class="lcui-swatch-item lcui-swatch-item--color"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="off"
      role="radio"
      style="--lcui-swatch-btn-border-color--active: rebeccapurple;"
      tabindex="-1"
      type="button"
    >
      <div
        class="lcui-swatch-color"
        style="background-color: rebeccapurple;"
      />
    </button>
    <button
      aria-checked="false"
      aria-label="gold"
      class="lcui-swatch-item lcui-swatch-item--color"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-state="off"
      role="radio"
      style="--lcui-swatch-btn-border-color--active: gold;"
      tabindex="-1"
      type="button"
    >
      <div
        class="lcui-swatch-color"
        style="background-color: gold;"
      />
    </button>
  </div>
</div>
`;
